classes/class3/README.md

```diff
@@ -1 +1,364 @@
+
+At the end of the day, templating engines-like most things-come down to personal preference, and that's why we're forcing you to like Handlebars! (At least for the next couple weeks).
+
+### Making Express Render Handlebars
+
+To get Express to render handlebars, as well as to add some useful templating features which we will get into later, we are going to use the [Express-Handlebars](https://github.com/ericf/express-handlebars) module. So go ahead and `npm --save install express-handlebars` to get it installed.
+
+Now two more things to know about are layouts and partials.
+* Layouts: Basically meta-templates. They are handlebars documents that allow you to create a wrapper around all the templates that use them.
+* Partials: Basically mini-reusable templates. They are handlebars documents that can be written once and "included" within other templates. (ei. a navigation bar that appears the same on every page)
+
+We're going to build a layout and leave utilizing partials as an exercise for later.
+
+To render a template the first thing we want is a base layout. To start we'll make a very simple one:
+```html
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>Express Intro</title>
+  </head>
+  <body>
+    {{{body}}}
+  </body>
+</html>
+```
+This is some of the boiler plate for HTML with the center bit `{{{body}}}`. That signifies where templates will be inserted into this layout.
+Go ahead and copy that code into a file `main.handlebars` and put it in a new folder `layouts` within the `views` directory of your application.
+
+Now for the template itself. Starting off simple:
+```html
+<h1>Welcome Home!</h1>
+```
+Save that in a file called `home.handlebars` and put in the `views` directory as a sibling to the new `layouts` directory you just made. Your folder structure should look like this:
+```
+.
+├── app.js
+└── views
+    ├── home.handlebars
+    └── layouts
+        └── main.handlebars
+
+2 directories, 3 files
+```
+All of this organization is completely customizable, but we'll leave it as the defaults for now.
+
+Now let's hook it all up!
+
+Add the line `var exphbs  = require('express-handlebars');` to the top of your `app.js` file. Then add these two lins after the `var app = express();` line:
+```javascript
+app.engine('handlebars', exphbs({defaultLayout: 'main'}));
+app.set('view engine', 'handlebars');
+```
+Note that you're setting the default layout for the application to the `main.handlebars` file you just created. You can change the layout for a single template at render time with a simple option.
+
+Lastly, within the `index.js` routing file, replace:
+```javascript
+res.send("Welcome home!");
+```
+with
+```javascript
+res.render('home');
+```
+Restart your server, visit http://localhost:3000 and behold your template!
```

Comment:
In class, this might be a good place to open up the dev console and show that the full template was sent over.